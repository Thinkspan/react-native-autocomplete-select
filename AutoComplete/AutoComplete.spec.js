/* global describe, it, expect, beforeEach, jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import AutoComplete from '../'

Enzyme.configure({ adapter: new Adapter() })

function noop () {}

describe('AutoComplete', () => {
  let fakeSuggestions
  beforeEach(() => {
    fakeSuggestions = [{
      text: '0.4'
    }, {
      text: '0.5'
    }, {
      text: '0.6'
    }]
  })

  it('renders correctly', () => {
    expect(shallow(
      <AutoComplete onSelect={noop} />
    )).toMatchSnapshot()
  })

  it('only renders the suggestions that have a similarity score greater than or equal to the minimum score passed', () => {
    const params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      minimumSimilarityScore: 0.5,
      onSelect: noop
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()
    wrapper.update()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    expect(wrapper.find({suggestionText: '0.5'}).length).toEqual(1)
    expect(wrapper.find({suggestionText: '0.6'}).length).toEqual(1)
  })

  it('correctly changes the rendered suggestions when receiving new props', () => {
    let params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      minimumSimilarityScore: 0.5,
      onSelect: noop
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()
    wrapper.update()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    params.value = '0.6'
    wrapper.setProps(params)

    expect(wrapper.find('TouchableHighlight').length).toEqual(1)
    expect(wrapper.find({suggestionText: '0.6'}).length).toEqual(1)
  })

  it('hides the suggestions and calls the select callback when a suggestion is selected', () => {
    const params = {
      value: '0.5',
      suggestions: fakeSuggestions,
      suggestionObjectTextProperty: 'text',
      onSelect: jest.fn(),
      minimumSimilarityScore: 0.5
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.instance().componentDidMount()
    wrapper.instance().forceUpdate()
    wrapper.update()

    expect(wrapper.find('TouchableHighlight').length).toEqual(2)

    wrapper.find({suggestionText: '0.5'}).simulate('press')

    expect(params.onSelect.mock.calls.length).toEqual(1)
    expect(params.onSelect.mock.calls[0]).toEqual([{
      text: '0.5'
    }])

    wrapper.instance().forceUpdate()
    wrapper.update()

    expect(wrapper.find('TouchableHighlight').length).toEqual(0)
  })

  it('correctly calls the onChangeText callback when the text changes', () => {
    const params = {
      onChangeText: jest.fn(),
      onSelect: noop
    }

    const wrapper = shallow(
      <AutoComplete {...params} />
    )

    wrapper.find('TextInput').simulate('changeText', 'Some Text')

    expect(params.onChangeText.mock.calls.length).toEqual(1)
    expect(params.onChangeText.mock.calls[0]).toEqual(['Some Text'])
  })
})
