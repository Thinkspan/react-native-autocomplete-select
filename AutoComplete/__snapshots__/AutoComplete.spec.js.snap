// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AutoComplete
    comparationFuzziness={0.5}
    minimumSimilarityScore={0.6}
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextInput
          comparationFuzziness={0.5}
          minimumSimilarityScore={0.6}
          onChangeText={[Function]}
          onSelect={[Function]}
          style={
            Object {
              "fontSize": 15,
            }
          }
        />,
        null,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "comparationFuzziness": 0.5,
          "minimumSimilarityScore": 0.6,
          "onChangeText": [Function],
          "onSelect": [Function],
          "style": Object {
            "fontSize": 15,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextInput
            comparationFuzziness={0.5}
            minimumSimilarityScore={0.6}
            onChangeText={[Function]}
            onSelect={[Function]}
            style={
              Object {
                "fontSize": 15,
              }
            }
          />,
          null,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "comparationFuzziness": 0.5,
            "minimumSimilarityScore": 0.6,
            "onChangeText": [Function],
            "onSelect": [Function],
            "style": Object {
              "fontSize": 15,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
